src/components/aeBanner/aeBanner.js

```diff
@@ -0,0 +1,67 @@
+const defaultOptions = {
+  timeout: 2500,
+  autoDestroy: true
+}
+
+export default {
+  name: 'ae-banner',
+  data: function () {
+    return {
+      show: false
+    }
+  },
+  props: {
+    'options': {
+      type: Object
+    }
+  },
+  mounted () {
+  },
+  methods: {
+    /**
+     * Show the banner
+     *
+     * @public
+     */
+    showBanner () {
+      if (this.show) {
+        return
+      }
+      this.show = true
+
+      if (this.allOptions.autoDestroy) {
+        this._startLazyDestroy()
+      }
+    },
+    /**
+     * Hide the banner
+     *
+     * @public
+     */
+    hideBanner () {
+      if (!this.show) {
+        return
+      }
+      this._clearTimer()
+      this.show = false
+    },
+    _startLazyDestroy () {
+      this.timers = setTimeout(() => {
+        this.hideBanner()
+      }, this.allOptions.timeout)
+    },
+    _clearTimer () {
+      if (this.timers) {
+        clearTimeout(this.timers)
+      }
+    }
+  },
+  computed: {
+    hasRightItem () {
+      return !!this.$slots.right
+    },
+    allOptions () {
+      return Object.assign({}, defaultOptions, this.options)
```

Comment:
Will be more clear to define two properties: `timeout` as Number and `auto-destroy` as Boolean with proper default values.

Reply:
Or one property `auto-destroy-timeout` and if it is not set then don't hide a banner.

Reply:
Having two separate properties would be well enough. As one may only wish to change the timeout.